pkg/yarn/copilot/server/server.go

```diff
@@ -0,0 +1,115 @@
+package server
+
+import (
+	"context"
+	"fmt"
+	"net"
+	"net/http"
+	"os"
+	"time"
+
+	"github.com/gin-gonic/gin"
+	v1 "k8s.io/api/core/v1"
+	"k8s.io/klog/v2"
+
+	"github.com/koordinator-sh/koordinator/pkg/yarn/copilot/nm"
+)
+
+type YarnCopilotServer struct {
+	mgr      *nm.NodeMangerOperator
+	unixPath string
+}
+
+func NewYarnCopilotServer(mgr *nm.NodeMangerOperator, unixPath string) *YarnCopilotServer {
+	return &YarnCopilotServer{mgr: mgr}
+}
+
+func (y *YarnCopilotServer) Run(ctx context.Context) error {
+	e := gin.New()
+	e.GET("/v1/container", y.GetContainer)
```

Comment:
these URI and protocols should be defined in yarn/copilot/api pkg for using in koordlet